iceoryx_hoofs/include/iceoryx_hoofs/internal/log/logstream.inl

```diff
@@ -86,66 +95,54 @@ inline LogStream& LogStream::self() noexcept
 inline LogStream& LogStream::operator<<(const char* cstr) noexcept
 {
     m_logger.logString(cstr);
-    m_flushed = false;
+    m_isFlushed = false;
     return *this;
 }
 
 inline LogStream& LogStream::operator<<(const std::string& str) noexcept
 {
     m_logger.logString(str.c_str());
-    m_flushed = false;
+    m_isFlushed = false;
     return *this;
 }
 
-template <typename T, typename std::enable_if_t<std::is_signed<T>::value, int>>
-inline LogStream& LogStream::operator<<(const T val) noexcept
+inline LogStream& LogStream::operator<<(const bool val) noexcept
 {
-    m_logger.logI64Dec(val);
-    m_flushed = false;
+    m_logger.logBool(val);
     return *this;
 }
 
-template <typename T, typename std::enable_if_t<std::is_unsigned<T>::value, int>>
+template <typename T, typename std::enable_if_t<std::is_arithmetic<T>::value, int>>
 inline LogStream& LogStream::operator<<(const T val) noexcept
 {
-    m_logger.logU64Dec(val);
-    m_flushed = false;
+    m_logger.logDec(val);
+    m_isFlushed = false;
     return *this;
 }
 
-template <typename T, typename std::enable_if_t<std::is_signed<T>::value, int>>
-inline LogStream& LogStream::operator<<(const LogHex<T>&& val) noexcept
+template <typename T, typename std::enable_if_t<std::is_integral<T>::value, int>>
+inline LogStream& LogStream::operator<<(const LogHex<T> val) noexcept
 {
     m_logger.logString("0x");
-    m_logger.logU64Hex(static_cast<uint64_t>(val.m_value));
-    m_flushed = false;
+    m_logger.logHex(static_cast<typename std::make_unsigned<T>::type>(val.m_value));
+    m_isFlushed = false;
     return *this;
 }
 
-template <typename T, typename std::enable_if_t<std::is_unsigned<T>::value, int>>
-inline LogStream& LogStream::operator<<(const LogHex<T>&& val) noexcept
+template <typename T, typename std::enable_if_t<std::is_floating_point<T>::value, int>>
+inline LogStream& LogStream::operator<<(const LogHex<T> val) noexcept
 {
-    m_logger.logString("0x");
-    m_logger.logU64Hex(val.m_value);
-    m_flushed = false;
-    return *this;
-}
-
-template <typename T, typename std::enable_if_t<std::is_signed<T>::value, int>>
-inline LogStream& LogStream::operator<<(const LogOct<T>&& val) noexcept
-{
-    m_logger.logString("0o");
-    m_logger.logU64Oct(static_cast<uint64_t>(val.m_value));
-    m_flushed = false;
+    m_logger.logHex(val.m_value);
+    m_isFlushed = false;
     return *this;
 }
 
-template <typename T, typename std::enable_if_t<std::is_unsigned<T>::value, int>>
-inline LogStream& LogStream::operator<<(const LogOct<T>&& val) noexcept
+template <typename T, typename std::enable_if_t<std::is_integral<T>::value, int>>
+inline LogStream& LogStream::operator<<(const LogOct<T> val) noexcept
```

Comment:
I wouldn't do it here. `std::cout << std::oct << -42 << std::endl;` also works and therefore I would keep it. In the console logger we use `snprintf` which does not support signed integer. One could of course argue to do the conversion within the console logger but since it is easy to add an API and hard to remove one I decided to leave it for the future if necessary.